app/models/vote.rb

```diff
@@ -0,0 +1,9 @@
+class Vote < ApplicationRecord
+  belongs_to :user
+  has_one :work
+  validates :user_id, presence: true
+  validates :work_id, presence: true
```

Comment:
Having the relationship means these validations are unnecessary.  However you can put one in to make sure the combination of user_id and work_id are unique.

```suggestion
  validates :user_id, uniqueness: { scope: :work_id }
```